### PR TITLE
Dot support for localField

### PR DESCRIPTION
**Pull request by superqwer**
related issue https://github.com/mongomock/mongomock/issues/595

### CODE REVIEW

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -601,6 +601,29 @@ def fixed_getter(doc):
     return fixed_getter
 
 
+def _get_local_field_val(dct, path):
```
Comment (pcorpet) [inline mongomock/aggregate.py]: Please look at `helpers.get_value_by_dot` and consider using it. Also please add tests for some corner cases covered by `get_value_by_dot` not covered by your code here.

File: tests/test__collection_api.py [hunk 2]
```diff
@@ -2464,18 +2464,33 @@ def test__aggregate_lookup_operator_not_string(self):
...
-                    'from': 'b',
-                    'localField': 'should.fail',
-                    'foreignField': 'arr',
-                    'as': 'b'
-                }}
-            ])
-        self.assertIn(
-            "Although '.' is valid in the 'localField' and 'as' parameters",
-            str(err.exception))
+        self.db.a.insert_many([
```
Comment (pcorpet) [inline tests/test__collection_api.py:2467]: Please add an equivalent test in `test__mongomock` to ensure that the feature is behaving the same as in `pymongo`

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks for the PR

**Comment by superqwer** [comment]
Changed the resolver to get_value_by_dot
Added mongomock test

File: tests/test__collection_api.py (hunk 2 above)
Comment (superqwer) [reply tests/test__collection_api.py:2467]: done

File: mongomock/aggregate.py (hunk 1 above)
Comment (superqwer) [reply mongomock/aggregate.py]: changed to get_value_by_dot
done

**Review by pcorpet — APPROVED** [review]
